torch/distributed/_tensor/device_mesh.py

```diff
@@ -172,8 +171,12 @@ def _init_process_groups(self):
         # validate that all calling ranks pass in the same `mesh` argument.
         self_mesh = self.mesh.to(self.device_type)
         mesh_list = [self_mesh.clone() for _ in range(get_world_size())]
-        all_gather(mesh_list, self_mesh)
-        for other_rank, other_mesh in enumerate(mesh_list):
+        mesh_tensor = torch.cat(mesh_list)
```

Comment:
nit: if uses functional collective, we don't need the `mesh_list` and the `mesh_tensor` cat, we can just do `funcol.all_gather_tensor(self_mesh, gather_dim=0)` instead.

Reply:
sg, ty :)